pkg/storage/s3.go

```diff
@@ -0,0 +1,84 @@
+package storage
+
+import (
+	"bytes"
+	"io/ioutil"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"golang.org/x/net/context"
+)
+
+const ()
+
+// newS3Backend returns a new backend implementation backed by the supplied
+// S3 bucket name. Note that with the current interface, the AWS_REGION environment
+// variable must be specified to use this backend.
+func newS3Backend(bucketName string) (*s3Backend, error) {
+	return &s3Backend{bucketName: aws.String(bucketName)}, nil
+}
+
+type s3Backend struct {
+	bucketName *string
+}
+
+func (b *s3Backend) read(_ context.Context, id string) ([]byte, error) {
+	sess := session.Must(session.NewSession())
```

Comment:
From the docs [1] it seems that these sessions (or perhaps even the client?) are designed to be cached/reused. Should we perahps store one in the backend object rather than creating it on every call? Here and below.

[1] https://docs.aws.amazon.com/sdk-for-go/api/aws/session/

Reply:
It seems like storing the client makes sense for now. That may change, but it's a detail that doesn't impact the interface.